app/controllers/generic_object_definition_controller.rb

```diff
@@ -23,6 +24,27 @@ def self.model
     GenericObjectDefinition
   end
 
+  def button
+    if params[:pressed] == 'generic_object_definition_new'
```

Comment:
nitpick: consider using case statement - something like (untested): 
```ruby
    javascript_redirect(
      case params[:pressed]
      when 'generic_object_definition_new'
        { :action => 'new' }
      when '￼generic_object_definition_edit'
        { :action => 'edit', :id => from_cid(params[:id] || params[:miq_grid_checks]) }
      end
    )
```

Reply:
Interesting construct.

I can make that change. Thanks.

Reply:
Done in f049138